server/app.js

```diff
@@ -0,0 +1,55 @@
+const express = require('express');
+const cors = require('cors');
+const bodyParser = require('body-parser');
+const Result = require('./models/search');
+const app = express();
+app.use(cors());
+app.use(bodyParser.json());
+
+
+
+
+app.get('/', (req, res) => {
+    res.send('Welcome!');
+});
+
+app.get('/results', (req, res) => {
+    const resultsData = Result.all
+    res.send(resultsData);
+});
+
+
+
+// Helpers
+// Normal Search Helpers
+const storedSearches = ['Banana', 'Fish', 'Couscous', 'Rice', 'Steak', 'Pasta', 'Fufu', 'Hamburger', 'Shawarma', 'Kimchi'];
+
+function filterSearch(query){
+    const resultsData = Result.all
+    const resultsFiltered = resultsData.filter((Rslt) => Rslt.query.toLowerCase() === query.toLowerCase())
+    return resultsFiltered
+}
+
+// Random Page Helper 
+function getRandomPage () {
+    const resultsData = Result.all
+    return (resultsData[Math.floor(Math.random()*resultsData.length)]);
+}
+
+
+// Generate GET for pages 
+storedSearches.map(function searchResults(query){
+    app.get('/results/' + query, (req, res) => res.send(filterSearch(query)))
+})
```

Comment:
Function searchResults doesn't return anything. Array.prototype.map requires it's callback to return.
.map returns an array, how did you plan to implement the routes?
Did you consider using the req.params ?


E.G  app.get('/result/:query', (req, res) => {
  const query = req.params.query;
  // process query
})